src/hyperv/mod.rs

```diff
@@ -109,9 +116,19 @@ pub mod report {
     }
 
     fn hcl_report(bytes: &[u8]) -> Result<AttestationReport> {
-        let hcl: Hcl =
-            bincode::deserialize(bytes).context("unable to deserialize bytes from vTPM")?;
+        if bytes.len() < TOTAL_SIZE {
+            return Err(anyhow!(
+                "HCL report size mismatch: expected at least {}, got {}",
+                TOTAL_SIZE,
+                bytes.len()
+            ));
+        }
+
+        let report_bytes = bytes
+            .get(REPORT_RANGE)
+            .ok_or_else(|| anyhow!("Failed to extract report bytes from HCL buffer"))?;
```

Comment:
Because you have already validated the byte length above, there isn't a need to use the `.get()` method here. You can optimize this operation by doing
```suggestion
        let report_bytes = &bytes[REPORT_RANGE]
```

Reply:
Fixed